src/Features/Core/Portable/CodeStyle2/AbstractCodeStyle2Provider.cs

```diff
@@ -0,0 +1,312 @@
+
+            public sealed override bool OpenFileOnly(Workspace workspace)
+                => _codeStyleProvider.DiagnosticsForOpenFileOnly(workspace);
+        }
+
+        #endregion
+
+        #region refactoring
+
+        private async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var optionValue = optionSet.GetOption(_option);
+
+            var severity = optionValue.Notification.Severity;
+            switch (severity)
+            {
+                case ReportDiagnostic.Suppress:
+                case ReportDiagnostic.Hidden:
+                    // if the severity is Hidden that's equivalent to 'refactoring only', so we want
+                    // to try to compute the refactoring here.
+                    //
+                    // If the severity is 'suppress', that means the user doesn't want the actual
+                    // analyzer to run here.  However, we can still check to see if we could offer
+                    // the feature here as a refactoring.
+                    await ComputeForwardRefactoringsAsync(context).ConfigureAwait(false);
+                    break;
+
+                case ReportDiagnostic.Error:
+                case ReportDiagnostic.Warn:
+                case ReportDiagnostic.Info:
+                    // User has this option set at a level where we want it checked by the
+                    // DiagnosticAnalyser and not the CodeRefactoringProvider.  However, we still
+                    // want to check if we want to offer the *reverse* refactoring here in this
+                    // single location.
+                    //
+                    // For example, say this is the "use expression body" feature.  If the user says
+                    // they always prefer expression-bodies (with warning level), then we want the
+                    // analyzer to always be checking for that.  However, we still want to offer the
+                    // refactoring to flip their code to use a block body here, just in case that
+                    // was something they wanted to do as a one off (i.e. before adding new
+                    // statements.
+                    //
+                    // TODO(cyrusn): Should we only do this for warn/info?  Argument could be made
+                    // that we shouldn't even offer to refactor in the reverse direction if it will
+                    // just cause an error.  That said, maybe this is just an intermediary step, and
+                    // we shouldn't really be blocking the user from making it.
+                    await ComputeReverseRefactoringsAsync(context).ConfigureAwait(false);
+                    break;
+            }
```

Comment:
@sharwell as above, i'm not sure what behavior to have for ReportDiagnostic.Default.  Can you help explain to me what that value means?